src/clients/c++/library/http_client.cc

```diff
@@ -1080,8 +1080,12 @@ InferenceServerHttpClient::Infer(
 
   // During this call SEND_END (except in above case), RECV_START, and
   // RECV_END will be set.
-  if (curl_easy_perform(easy_handle_) != CURLE_OK) {
+  auto curl_status = curl_easy_perform(easy_handle_);
+  if (curl_status != CURLE_OK) {
     sync_request->http_code_ = 400;
+    if (curl_status == CURLE_OPERATION_TIMEDOUT) {
```

Comment:
I would use cout and conditionalize with the verbose flag since it is not necessarily an error when a timeout occurs.